### ISSUE TITLE
Support log4j2

### ISSUE THREAD

**Issue by ralf0131** [apache/incubator-dubbo#1713]
Need to add the log4j2 support, currently supported logging framework:
* slf4j
* jcl
* log4j
* jdk

I think the priority is between low and medium.

**Comment by ralf0131** [comment]
Meanwhile, the English documentation[1] has only described the default value, and is missing all supported values, compared to the Chinese documentation[2].

[1] http://dubbo.apache.org/books/dubbo-user-book-en/references/xml/dubbo-application.html
[2] http://dubbo.apache.org/books/dubbo-user-book/references/xml/dubbo-application.html

**Comment by zhangzhxb520** [comment]
I found code in *LoggerFactory*，Why use log4j as default logger when system property "dubbo.application.logger" not set.

```java
static {
        String logger = System.getProperty("dubbo.application.logger");
        if ("slf4j".equals(logger)) {
            setLoggerAdapter(new Slf4jLoggerAdapter());
        } else if ("jcl".equals(logger)) {
            setLoggerAdapter(new JclLoggerAdapter());
        } else if ("log4j".equals(logger)) {
            setLoggerAdapter(new Log4jLoggerAdapter());
        } else if ("jdk".equals(logger)) {
            setLoggerAdapter(new JdkLoggerAdapter());
        } else {
            try {
                setLoggerAdapter(new Log4jLoggerAdapter());
            } catch (Throwable e1) {
                try {
                    setLoggerAdapter(new Slf4jLoggerAdapter());
                } catch (Throwable e2) {
                    try {
                        setLoggerAdapter(new JclLoggerAdapter());
                    } catch (Throwable e3) {
                        setLoggerAdapter(new JdkLoggerAdapter());
                    }
                }
            }
        }
    }
```

**Comment by ralf0131** [comment]
This issue has been fixed from 2.7.0 onwards.
